rpc/connection.go

```diff
@@ -83,12 +84,27 @@ func NewConnectionTransport(uri *FMPURI, l LogFactory, wef WrapErrorFunc, maxFra
 	}
 }
 
-func (t *connTransport) Dial(context.Context) (Transporter, error) {
+// NewConnectionTransportWithDialable creates a ConnectionTransport for a given FMPURI via the given Dialable
+func NewConnectionTransportWithDialable(uri *FMPURI, l LogFactory, wef WrapErrorFunc, maxFrameLength int32, dialable Dialable) ConnectionTransport {
+	return &connTransport{
+		uri:            uri,
+		l:              l,
+		wef:            wef,
+		maxFrameLength: maxFrameLength,
+		dialable:       dialable,
+	}
+}
+
+func (t *connTransport) Dial(ctx context.Context) (Transporter, error) {
 	var err error
 	if t.conn != nil {
 		t.conn.Close()
 	}
-	t.conn, err = t.uri.Dial()
+	if t.dialable == nil {
+		t.conn, err = t.uri.Dial()
```

Comment:
I'd prefer this to be:
```go
if t.dialable != nil {
	t.conn, err = t.dialable.Dial(ctx, "tcp", t.uri.HostPort)
} else {
	t.conn, err = t.uri.Dial()
}
```

That way, the `t.dialable` check is paired with the `t.dialable.Dial` call.